internal/edgedbtypes/datetime.go

```diff
@@ -660,6 +660,26 @@ func (d Duration) String() string {
 	return strings.Join(buf, "")
 }
 
+// AsNanoseconds returns [time.Duration] represented as nanoseconds,
+// after transforming from Duration microsecond representation.
+// Returns an error if the Duration is too long and would cause an overflow of
+// the internal int64 representation.
+func (d Duration) AsNanoseconds() (time.Duration, error) {
+	if int64(d) > math.MaxInt64/int64(time.Microsecond) ||
+		int64(d) < math.MinInt64/int64(time.Microsecond) {
+		return time.Duration(0), fmt.Errorf(
+			"Duration is too large to be represented as nanoseconds",
+		)
+	}
+	return time.Duration(d) * time.Microsecond, nil
+}
+
+// FromNanoseconds creates a Duration represented as microseconds
+// from a [time.Duration] represented as nanoseconds.
+func FromNanoseconds(d time.Duration) Duration {
+	return Duration(d.Microseconds())
```

Comment:
I was rereading the documentation and rediscovered that clients are supposed to [round to the nearest even](https://www.edgedb.com/docs/reference/bindings/datetime#precision).

There is a reference implementation [here](https://github.com/edgedb/edgedb-go/blob/0be826dd40630c314c470cc0c2b35fccb4b787de/internal/codecs/datetime.go#L91).

Reply:
Is using `math.RoundToEvent` acceptable ? I find it more straightforward
```go
func FromNanoseconds(d time.Duration) Duration {
	return Duration(math.RoundToEven(float64(d) / 1e3))
}
```

Reply:
Yea! I had no idea that existed :exploding_head: